### PR TITLE
feat(client): preview feature deno

### PR DESCRIPTION
**Pull request by millsp**
Integration version of https://github.com/prisma/prisma/pull/14386

### CODE REVIEW

**Comment by janpio** [comment]
(Sorry, I tried to resolve the conflicts and had messed up - so had to force push the previous state again. No changes.)

File: packages/cli/src/Generate.ts [hunk 1]
```diff
@@ -246,8 +255,18 @@ ${chalk.dim('```')}
...
+        } else {
+          hint = `You can now start using Prisma Client in your code. Reference: ${link('https://pris.ly/d/client')}
+${chalk.dim('```')}
+${highlightTS(`\
+import { PrismaClient } from '${importPath}/deno/edge.ts'`)}
+${chalk.dim('```')}
+
+You will need a Prisma Data Proxy connection string. See documentation: ${link('https://pris.ly/d/data-proxy')}
+${breakingChangesStr}${versionsWarning}`
+        }
```
Comment (janpio) [inline packages/cli/src/Generate.ts:269]: This feels a bit meh that deno is the `else` clause. Can we not react to `isDeno` here as well?

File: packages/client/src/generation/TSClient/TSClient.ts [hunk 2]
```diff
@@ -66,6 +67,7 @@ export class TSClient implements Generatable {
       runtimeName,
       datasources,
       dataProxy,
+      deno,
```
Comment (SevInf) [inline packages/client/src/generation/TSClient/TSClient.ts:70]: Question: how does `--deno` and `--data-proxy` interact with each other? Does `--deno` imply `--data-proxy`? Or are they incompatible? If latter, will we have an error if they are both specified?

**Review by SevInf — APPROVED** [review]
LGTM with a small question

File: packages/client/src/generation/TSClient/TSClient.ts (hunk 2 above)
Comment (janpio) [reply packages/client/src/generation/TSClient/TSClient.ts:70]: `--deno` does not exist, only `deno` in the list of preview features in the schema.

Comment (SevInf) [reply packages/client/src/generation/TSClient/TSClient.ts:70]: Got it, thanks

Comment (millsp) [reply packages/client/src/generation/TSClient/TSClient.ts:70]: The Deno CLI is detected, on top of that. So 3 requirements at the moment

**Pull request merged**